event-store-management/event-store-management-core/src/main/java/uk/gov/justice/services/eventstore/management/catchup/process/EventSourceNameFilter.java

```diff
@@ -0,0 +1,30 @@
+package uk.gov.justice.services.eventstore.management.catchup.process;
+
+import uk.gov.justice.subscription.domain.subscriptiondescriptor.Subscription;
+
+import java.util.List;
+import java.util.Optional;
+
+import javax.inject.Inject;
+
+public class EventSourceNameFilter {
+
+    @Inject
+    private CatchupWhitelistedEventSourcesProvider catchupWhitelistedEventSourcesProvider;
+
+    /**
+     * Will return true if either the event-source name is found in our JNDI list of whitelisted event sources
+     * or if the JNDI value has not been set
+     *
+     * @param subscription The Subscription that contains the name of an event source used for catchup
+     * @return true if the event-source name is found in our JNDI list of whitelisted event sources
+     */
+    public boolean shouldRunCatchupAgainstEventSource(final Subscription subscription) {
```

Comment:
is should the right naming convention for not test methods?

Reply:
I did think about that name a lot. The idea was to try to make the calling code understandable

 >  subscriptionsDescriptor.getSubscriptions()
                .stream()
                .filter(subscription -> eventSourceNameFilter.shouldRunCatchupAgainstEventSource(subscription))
                .sorted(priorityComparatorProvider.getSubscriptionComparator())
                
             
 Though if you can think of a better name let me know